executables/collector/src/main/java/de/cyface/collector/handler/MeasurementHandler.java

```diff
@@ -95,24 +99,46 @@ public final class MeasurementHandler implements Handler<RoutingContext> {
     private final long payloadLimit;
 
     /**
+     * Creates a fully initialized instance of this class.
+     *
+     * @param config the configuration setting required to start the HTTP server
+     */
+    public MeasurementHandler(final Config config) {
+        this(config.getDataDatabase(), config.getAuthProvider(), config.getUserDatabase(),
+                config.getMeasurementLimit());
+    }
+
+    /**
+     * Creates a fully initialized instance of this class.
+     *
      * @param dataClient The client to use to access the Mongo database.
+     * @param authProvider An auth provider used by this server to authenticate against the Mongo user database
+     * @param userDatabase The Mongo user database containing all information about users
      * @param payloadLimit The maximum number of {@code Byte}s which may be uploaded.
      */
-    public MeasurementHandler(final MongoClient dataClient, final long payloadLimit) {
+    public MeasurementHandler(final MongoClient dataClient, final MongoAuthentication authProvider,
+            final MongoClient userDatabase, final long payloadLimit) {
+        super(authProvider, userDatabase, true);
         Validate.notNull(dataClient);
         Validate.isTrue(payloadLimit > 0);
         this.dataClient = dataClient;
         this.payloadLimit = payloadLimit;
     }
 
     @Override
-    public void handle(final RoutingContext ctx) {
+    protected void handleAuthorizedRequest(final RoutingContext ctx, final List<de.cyface.api.model.User> users,
```

Comment:
It seems weird to provide the userDatabase in the constructor but also inject a list of users into the handleAuthorizedRequest method. Why not either load the users (on demand) directly from the database or inject the remaining data preloaded into the constructor. To remove the hard dependency on Mongo client code, you could introduce a layer of abstraction.